### PR TITLE
test(registration): add integration tests for registration

### LINKED ISSUE #177: Bug: Admins kan ikke melde folk av arrangementer
Det går ikke å melde brukere av arrangementer i admin-siden på `tihlde.org/admin/arrangementer/`. En DELETE-request til `/event/<event_id>/users/<user_id>` returnerer feilkode 400, se bilder:

image
image

### PR DESCRIPTION
**Pull request by Eirsteir**
# Add integration tests for registration
## Implement integratin tests for registrations, refactored event integration tests a little as well as a (minor) refactor of RegistrationViewSet
Link to Trello task: #113, closes #177

Comments/issues:

Tried not to introduce too many changes, only what was strictly necessary. Planning on a refactor of the viewset once this is through.

Changes: 
- create returns registration on success
- update returns registration on success
- admins can delete registrations after sign off deadline

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] PR name is descriptive
- [x] The PR includes the link to the relevant Trello task
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

- [x] `python manage.py runserver` was run locally witout errors

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

**Review by Zenjjim — CHANGES_REQUESTED** [review]
Beautiful work. Really enjoying how easy it is to read these tests. A few comments that should be easy to fix :D

File: app/content/views/registration.py [hunk 1]
```diff
@@ -61,23 +60,23 @@ def create(self, request, event_id):
...
             if serializer.is_valid():
-                Registration(user=user, event=event).save()
-                return Response({"detail": "Registration created."}, status=200)
+                registration = Registration.objects.create(user=user, event=event)
+                registration.save()
+                registration.refresh_from_db()
+                registration_serializer = RegistrationSerializer(
+                    registration, context={"user": registration.user}
+                )
+                return Response({"detail": registration_serializer.data}, status=201)
```
[the remark this answers is not in the document]
Comment (olros) [reply app/content/views/registration.py:69]: Stemmer, enten data eller detail: string

File: app/content/views/registration.py [hunk 2]
```diff
@@ -61,23 +60,23 @@ def create(self, request, event_id):
             user = User.objects.get(user_id=request.id)
             serializer = RegistrationSerializer(data=request.data)
             if serializer.is_valid():
-                Registration(user=user, event=event).save()
-                return Response({"detail": "Registration created."}, status=200)
+                registration = Registration.objects.create(user=user, event=event)
+                registration.save()
+                registration.refresh_from_db()
+                registration_serializer = RegistrationSerializer(
```
[the remark this answers is not in the document]
Comment (Eirsteir) [reply app/content/views/registration.py:66]: Det var primært for å serialize den nye registreringen, men skal se om det ikke går fint uten 👍

[the remark this answers is not in the document]
Comment (Eirsteir) [reply app/content/views/registration.py:66]: Inputet til serializeren er på feil format, så grunnen er at vi vil serialize det nye Registration-objektet. Testene feiler hvis man bare returnerer den "gamle" serializeren, `serializer.data`

File: app/tests/test_registration_integration.py [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+    client = get_api_client(user=member)
+    url = _get_registration_url(registration.event)
+    response = client.get(url)
+
+    assert response.status_code == 403
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "group_name", [AdminGroup.HS, AdminGroup.DEVKOM,],
```
[the remark this answers is not in the document]
Comment (Eirsteir) [reply app/tests/test_registration_integration.py:59]: Ja

[the remark this answers is not in the document]
Comment (Eirsteir) [reply app/tests/test_registration_integration.py:59]: Gjelder det for alle http-metoder også? Ser for meg at de burde ha tilgang til å administrere det

File: app/content/enums.py [hunk 4]
```diff
@@ -22,3 +22,4 @@ class AdminGroup(ChoiceEnum):
     DEVKOM = "DevKom"
```
Comment (Zenjjim) [inline app/content/enums.py:22]: Chi heter index nå 😊

**Review by Zenjjim — APPROVED** [review]
Nice and clean. A couple of comments for further issues

File: app/content/enums.py (hunk 4 above)
Comment (olros) [reply app/content/enums.py:22]: Ohh, hvis det endres i backend så må det endres i frontend også så vi får tilgang til admin-panelet

Comment (Eirsteir) [reply app/content/enums.py:22]: Fikser du det @olros?

Comment (olros) [reply app/content/enums.py:22]: Jepp 👍

**Pull request merged**